### PR TITLE
Doc warn and warn_with

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/types.jl [hunk 1]
```diff
@@ -849,6 +853,15 @@ end
 
 with(f::Function, ::Type{T}, args...) where {T} = with(f, T(args...))
 
+"""
+    with_warn(f::Function, ::Type{T}, args..)
```
Comment (ararslan) [inline base/libgit2/types.jl]: Missing a trailing dot on `args`

File: base/libgit2/types.jl [hunk 2]
```diff
@@ -849,6 +853,15 @@ end
...
+
+Resource management helper function. Apply `f` to `args`, first constructing
+an instance of type `T` out of them. Makes sure to call `close` on the resulting
```
Comment (ararslan) [inline base/libgit2/types.jl]: "Out of them" sounds confusing on first read. Maybe "first constructing an instance of type `T` using the given arguments"?

File: base/libgit2/types.jl (hunk 1 above)
Comment (kshyatt) [reply base/libgit2/types.jl]: oooh good call

**Review by ararslan — APPROVED** [review]

**Comment by kshyatt** [comment]
Looks like we had a Mac timeout. I find the CircleCI logs really hard to read - is the fail there my fault?

**Comment by ararslan** [comment]
The Circle logs never even load for me (as an aside, that's why I made #23417, though that seems unlikely to be merged), so I'm also not sure. But I don't think Circle runs the doc stuff so it's likely unrelated.

**Pull request merged**